app/models/spree/variant.rb

```diff
@@ -64,7 +64,7 @@ class Variant < ApplicationRecord
     validates_lengths_from_database
     validate :check_currency
     validates :price, numericality: { greater_than_or_equal_to: 0 }, presence: true
-    validates :tax_category, presence: true,
+    validates :tax_category_id, presence: true,
```

Comment:
That's a shame. This format has a few downsides, for example a new tax category can't be associated to a variant before it's saved. So instead of letting Rails save the record, we have to save it first. That shouldn't be in issue in our model but may be for test data.

Reply:
Yes, I thought it was a shame too, but I thought it's also more compatible with rails forms which is a good thing.
The test data issue is the only downside I can think of, can you think of any others?